src/include/common/types/value.h

```diff
@@ -154,171 +324,271 @@ class RelVal {
     std::vector<std::pair<std::string, std::unique_ptr<Value>>> properties;
 };
 
+/**
+ * @return the Value as a boolean value.
```

Comment:
If this template is recognized by Doxygen, then let's comment as `return boolean value`

Otherwise I think we don't need to comment specialized template